test/SgidClient.spec.ts

```diff
@@ -0,0 +1,332 @@
+import { readFileSync } from 'fs'
+
+import SgidClient from '../src'
+
+import {
+  MOCK_ACCESS_TOKEN,
+  MOCK_API_VERSION,
+  MOCK_AUTH_CODE,
+  MOCK_AUTH_ENDPOINT,
+  MOCK_CLIENT_ID,
+  MOCK_CLIENT_PRIVATE_KEY,
+  MOCK_CLIENT_SECRET,
+  MOCK_HOSTNAME,
+  MOCK_REDIRECT_URI,
+  MOCK_SUB,
+  MOCK_USERINFO_PLAINTEXT,
+} from './mocks/constants'
+import {
+  tokenHandlerNoSub,
+  tokenHandlerNoToken,
+  userInfoHandlerMalformedData,
+  userInfoHandlerMalformedKey,
+  userInfoHandlerNoData,
+  userInfoHandlerNoKey,
+} from './mocks/handlers'
+import { server } from './mocks/server'
+
+const DEFAULT_SCOPE = 'myinfo.nric_number openid'
+const DEFAULT_RESPONSE_TYPE = 'code'
+
+describe('SgidClient', () => {
+  let client: SgidClient
+
+  beforeEach(() => {
+    client = new SgidClient({
+      clientId: MOCK_CLIENT_ID,
+      clientSecret: MOCK_CLIENT_SECRET,
+      privateKey: MOCK_CLIENT_PRIVATE_KEY,
+      redirectUri: MOCK_REDIRECT_URI,
+      hostname: MOCK_HOSTNAME,
+      apiVersion: MOCK_API_VERSION,
+    })
+  })
+
+  describe('constructor', () => {
+    it('should initialise correctly when a PKCS1 private key is provided', () => {
+      expect(client).toBeDefined()
+    })
+
+    it('should initialise correctly when a PKCS8 private key is provided', () => {
+      const pkcs8Key = readFileSync(
+        `${__dirname}/mocks/mockPrivateKeyPkcs8.pem`,
+      ).toString()
+
+      const client = new SgidClient({
```

Comment:
The naming convention is confusing for scoped client and global client. May not be maintainable in long run.

```suggestion
      const pkcs8Client = new SgidClient({
```

Reply:
[a8f7374](https://github.com/opengovsg/sgid-client/pull/24/commits/a8f737474b342dca211714bb454fa727f299ebf4)